client/app/pages/data-sources/DataSourcesList.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import Button from 'antd/lib/button';
+import { react2angular } from 'react2angular';
+import { isEmpty, get } from 'lodash';
+import settingsMenu from '@/services/settingsMenu';
+import { DataSource, IMG_ROOT } from '@/services/data-source';
+import { policy } from '@/services/policy';
+import navigateTo from '@/services/navigateTo';
+import { $route } from '@/services/ng';
+import { routesToAngularRoutes } from '@/lib/utils';
+import CardsList from '@/components/cards-list/CardsList';
+import LoadingState from '@/components/items-list/components/LoadingState';
+import CreateSourceDialog from '@/components/CreateSourceDialog';
+import helper from '@/components/dynamic-form/dynamicFormHelper';
+
+class DataSourcesList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { dataSourceTypes: [], dataSources: [], loading: true };
```

Comment:
My personal pref is that if state has no initial values that rely on props, use the "out of constructor" syntax cause it's cleaner.

Reply:
Adopted :+1: